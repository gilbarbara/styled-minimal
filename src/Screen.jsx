import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { minHeight, space } from 'styled-system';

export const StyledScreen = styled.div`
  ${minHeight};
  ${space};
`;

const Screen = ({ children, ...props }) => (
  <StyledScreen {...props}>{children}</StyledScreen>
);

Screen.propTypes = {
  children: PropTypes.node.isRequired,
  minHeight: PropTypes.string,
};

Screen.defaultProps = {
  minHeight: '100vh',
};

export default Screen;
