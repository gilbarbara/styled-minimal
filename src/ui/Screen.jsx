import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { getProp } from '../utils/helpers';

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  min-height: 100vh;
  padding-top: ${getProp('gutter')};
`;

const Path = ({ children }) => (
  <Wrapper>{children}</Wrapper>
);

Path.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Path;
