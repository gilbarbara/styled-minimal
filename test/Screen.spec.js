import React from 'react';
import { mount } from 'enzyme';

import Screen from '../src/Screen';

describe('Screen', () => {
  const wrapper = mount(
    <Screen>Far far away</Screen>
  );

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
