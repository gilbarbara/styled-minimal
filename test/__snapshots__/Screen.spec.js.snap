// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen should render properly 1`] = `
.c0 {
  box-sizing: border-box;
  min-height: 100vh;
}

<Screen
  minHeight="100vh"
>
  <div
    className="c0"
  >
    Far far away
  </div>
</Screen>
`;
